File: Python3/0070_climbing_stairs.py
# https://leetcode-cn.com/problems/climbing-stairs/

# Let f(n) denote number of ways to climb to the nth stair
# with n = 0,1,....
# Then we have f(n) = f(n-1) + f(n-2).
# Specifically, f(0) = f(1) = 1.

class Solution:
    def climbStairs(self, n:int) -> int:
        res = [1,1]
        for i in range(2,n+1):
            res.append(res[i-1]+res[i-2])
        return res[n]

File: Cpp/0070_climbing_stairs.cpp
// https://leetcode.cn/problems/climbing-stairs/

class Solution {
public:
    int climbStairs(int n) {
        int last = 1, last_last = 1;
        int res = 1;
        for (int i = 1; i < n; ++i) {
            res = last + last_last;
            last_last = last;
            last = res;
        }
        return res;
    }
};
